account-kit/smart-contracts/src/ma-v2/account/account.ts

```diff
@@ -49,6 +50,8 @@ export type CreateSMAV2AccountAccountParams<
   initialOwner?: Address;
   accountAddress?: Address;
   entryPoint?: EntryPointDef<TEntryPointVersion, Chain>;
+  isGlobalValidation?: boolean;
```

Comment:
💬  I always question when booleans are passed around. In this case the global validation is just for the nonce creation in the `getAccountNonce` right. And At these points I think of other potential apis, like maybe the name could have been NonceBytes, with constants of GLOBAL_VALIDATION_NONCE = 1n, NO_OPS_NONCE = 0n. Or another could have been nonceModifiers (a: nonce) => nonce, basically a fn instead. I don't know what the correct choice is, but every time I see a flag param, I have these questions.

Reply:
Very thought provoking!

This current flag flips a bit in a byte in `getAccountNonce`, but there would eventually be another flag (`isDeferredValidation`) that toggles a second bit in the same byte. I was thinking of using the two booleans to create the final byte like this: `isDeferredAction ? 1 << 8 : 0 | isGlobalValidation ? 1 : 0`

Reply:
Which implementation would you choose for the above?

Reply:
So, funny enough, I might still do the boolean flags. 
I am trying to create a good thought provoking.
Given that I would probably choose the easiest like has been done, the other would be the fixed point operator as a function would still work. 
```ts
nonceModifier: NonceModifiers.None,
nonceModifier: NonceModifiers.DefferedAction,
nonceModifier: NonceModifiers.GlobalValidation,
nonceModifier: NonceModifiers.compose(NonceModifiers.GlobalValidation, NonceModifiers.DefferedAction),
nonceModifier: NonceModifiers.GlobalValidationAndDefferedAction,
```
I would go away from the single bit operator. But again, I'm just trying to give another tool for the toolbox. 
The problem with the isDefferedAction is seen that we add another flag, and we may add another in the future. If we had two places where the flags where being used, and a new flag was added. The chances of missing the other location for the flag is there. While this fn style lets us do the same operation in multiple areas.

And if we wanted this to be a typed with phantom, we could have done
```ts
nonceModifier: NonceModifier

type NonceModifier = <N extends  bigint & {__type: "NONCE"}>(nonce: N) => N & {__type: "MODIFED_NONCE}
/// Implement the type for each of the NonceModifiers
NonceModifiers.compose = (...as: NonceModifiers) => as.reduce((a, b) => (nonce) => a(b(nonce)), NonceModifiers.None) as NonceModifier
```